src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/Attribute/MetricFilter.php

```diff
@@ -0,0 +1,245 @@
+    protected $measureManager;
+
+    /** @var MeasureConverter */
+    protected $measureConverter;
+
+    /**
+     * @param AttributeValidatorHelper $attrValidatorHelper
+     * @param MeasureManager           $measureManager
+     * @param MeasureConverter         $measureConverter
+     * @param array                    $supportedAttributeTypes
+     * @param array                    $supportedOperators
+     */
+    public function __construct(
+        AttributeValidatorHelper $attrValidatorHelper,
+        MeasureManager $measureManager,
+        MeasureConverter $measureConverter,
+        array $supportedAttributeTypes = [],
+        array $supportedOperators = []
+    ) {
+        $this->attrValidatorHelper = $attrValidatorHelper;
+        $this->measureManager = $measureManager;
+        $this->measureConverter = $measureConverter;
+        $this->supportedAttributeTypes = $supportedAttributeTypes;
+        $this->supportedOperators = $supportedOperators;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addAttributeFilter(
+        AttributeInterface $attribute,
+        $operator,
+        $value,
+        $locale = null,
+        $channel = null,
+        $options = []
+    ) {
+        if (null === $this->searchQueryBuilder) {
+            throw new \LogicException('The search query builder is not initialized in the filter.');
+        }
+
+        $this->checkLocaleAndChannel($attribute, $locale, $channel);
+
+        if (Operators::IS_EMPTY !== $operator && Operators::IS_NOT_EMPTY !== $operator) {
+            $this->checkValue($attribute, $value);
+            $value = $this->convertValue($attribute, $value);
+        }
+
+        $attributePath = $this->getAttributePath($attribute, $locale, $channel) . '.' . self::PATH_SUFFIX;
+
+        switch ($operator) {
+            case Operators::LOWER_THAN:
+                $clause = [
+                    'range' => [
+                        $attributePath => ['lt' => $value]
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::LOWER_OR_EQUAL_THAN:
+                $clause = [
+                    'range' => [
+                        $attributePath => ['lte' => $value]
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::EQUALS:
+                $clause = [
+                    'term' => [
+                        $attributePath => $value
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_EQUAL:
+                $mustNotClause = [
+                    'term' => [
+                        $attributePath => $value,
+                    ],
+                ];
+                $filterClause = [
+                    'exists' => [
```

Comment:
Is it useful to search against the attribute path + the path prefix for `exists`?
I think that just the attribute path is enough.